materialize-redshift/sqlgen.go

```diff
@@ -2,23 +2,17 @@ package main
 
 import (
 	"fmt"
-	"regexp"
 	"slices"
 	"strings"
+	"text/template"
 	"time"
 	"unicode/utf8"
 
 	sql "github.com/estuary/connectors/materialize-sql"
 	"github.com/estuary/flow/go/protocols/fdb/tuple"
 )
 
-// Identifiers matching the this pattern do not need to be quoted. See
-// https://docs.aws.amazon.com/redshift/latest/dg/r_names.html. Identifiers (table names and column
-// names) are case-insensitive, even if they are quoted. They are always converted to lowercase by
-// default.
-var simpleIdentifierRegexp = regexp.MustCompile(`(?i)^[a-z_][a-z0-9_]*$`)
```

Comment:
Previously we wouldn't quote column names with uppercase characters, which never really made any difference since we always assumed that column names would be treated as lowercase. Now with the potential for handling case sensitivity, we can use the standard `sql.IsSimpleIdentifier` method which will quote identifiers with uppercase characters.